perfkitbenchmarker/pkb.py

```diff
@@ -509,6 +518,42 @@ def RunBenchmark(spec, collector):
   spec.status = benchmark_status.SUCCEEDED
 
 
+def RunBenchmarkTask(spec, collector):
+  """Task that executes RunBenchmark.
+
+  This is designed to be used with RunParallelProcesses.
+
+  Arguments:
+    spec: BenchmarkSpec. The spec to call RunBenchmark with.
+    collector: SamplesCollector. The collector to call RunBenchmark with.
+
+  Returns:
+    A tuple of BenchmarkSpec, list of samples.
+  """
+  if _TEARDOWN_EVENT.is_set():
+    return spec, []
+  # Many providers name resources using run_uris. When running multiple
+  # benchmarks in parallel, this causes name collisions on resources.
+  # By modifying the run_uri, we avoid the collisions.
```

Comment:
Not any more often than they could before - this guarantees that each benchmark in a run will have a different run uri. The chance of runs having the same base run uri is the same as before this change.